sphinx/config.py

```diff
@@ -626,27 +626,38 @@ def correct_copyright_year(_app: Sphinx, config: Config) -> None:
     See https://reproducible-builds.org/specs/source-date-epoch/
     """
     if source_date_epoch := int(getenv('SOURCE_DATE_EPOCH', '0')):
-        source_date_epoch_year = time.gmtime(source_date_epoch).tm_year
+        replace_year = time.gmtime(source_date_epoch).tm_year
     else:
-        return
```

Comment:
> I think we should only perform _any_ kind of substitution on the project copyright config when `SOURCE_DATE_EPOCH` configured.

See the change to `doc/conf.py` though -- the idea is that projects can now use static values in the config (which is a step towards `.toml` config files).

In my mind there are two paths:

1. New-style (%Y / %y): Always substitute with a year. Use the current year by default, and SOURCE_DATE_EPOCH otherwise.
2. Old style (no template): Substitute iff (SOURCE_DATE_EPOCH is set, the replace year matches the system year, and the replace year isn't in the future)

(There is a bug in the current PR that you've made me realise -- new-style should always be substituted, but this exits early if SOURCE_DATE_EPOCH doesn't meet conditions.)